### DISCUSSION TITLE
Auto update placed library components

### DISCUSSION THREAD

**Discussion by neatlittlemacapps** [six7/figma-tokens#202] [Ideas]
Placing components from a Central DS into a file with the same tokens but different values does not update automatically.
I have to click “update” everytime I drop a component in, in order for the newly placed components to update themselves to match the new token values from the file.

The usecase here would be to have a single Design System that has the main components defined.
This System would be used by 4 different “Products”. The “mechanics” of the components would have to be the same for all of the teams, but the “look” of the components not so. There **would not** be the freedom to tweak the functionality of these components. If this is needed, we’d built that into the component variants.

But **we would** allow flexibility for the teams to tweak their token values (not the tokens themselves)
I have been playing with this and I notice that when I place a component from the main library into a file that has a different token set (same tokens, different values) they do not update automatically to match the tokens. I have to click “update” in order for the newly placed components to update themselves to match the new tokens.

Unless the idea would be to use the token sets at the top, but I don’t think that’s what they’re there for. It would also make creating the initial DS quite heavy (having to switch between them and seeing the entire library switch).

**Comment by six7** [comment]
Let me recap so that we're on the same page: You have a main DS with tokens and components and everything setup, and you have another white label that is styled differently but also has a local token system. When users paste components from DS A to DS B they should update their token values (and they do once you hit Update). But it would be more convenient if the change of style happened as soon as users pasted the components over, right?

I think the main problem with this is the performance impact this would bring: We'd essentially have to listen to every selection change (which would get triggered if you Cmd + V). However, if on every selection change we update all layers with the new token values the performance impact would be quite huge, which is why I think forcing users to hit Update is acceptable in comparison.

Let me know if I misunderstood you and thanks for the feedback!

**Reply by neatlittlemacapps** [reply]
Kind of...
It's not upon copy/pasting but upon placing a component from the whitelabel library.
I included a video to explain it a bit better.

https://user-images.githubusercontent.com/60575307/129684078-3e29c15d-e100-463a-93d3-5cf9583c048f.mp4

I'm not sure if that drag/drop is the same as what you mean with "selection change"

Hope this is a bit more clear?

**Reply by six7** [reply]
Would it help your teams to have a quick way to do that via the command palette? Like a `Run Update` function your could run by entering `Cmd + P`? That way, the plugin wouldn't have to be open all the time, and whenever your team placed components it would run that (without having to start the plugin).

Because right now the only way to do what you're after is by performing that for every selection change (everytime you select something in Figma), and that will cost performance. Unfortunately there is no event in Figma we can listen to that would trigger when you place something from a library... Here's everything we get: https://www.figma.com/plugin-docs/api/properties/figma-on/

**Reply by neatlittlemacapps** [reply]
I understand the predicament. You can't go triggering on "The current page changed" because that would be triggered all of the time I guess. I'd have to check for the command palette solution. It does sound a bit like a hack though.

**Reply by georgeorgeorge** [reply]
I'm experiencing the exact same issue. It's the final hurdle for me to get it to re-brand upon dropping the component into the file.

Was a solution ever found for this? I wondered if utilising the 'apply to selection' option to run an update automatically would be a slightly more efficient way of doing this. And would [onDrop](https://www.figma.com/plugin-docs/api/properties/figma-on/#drop) be able to do this action?

Any advice or help would be massively appreciated 🙏🏼
